### ISSUE TITLE
Rule proposal: no inline callbacks before other parameters

### ISSUE THREAD

**Issue by fregante** [sindresorhus/eslint-plugin-unicorn#2234]
### Description

Initially reported in https://github.com/sindresorhus/memoize/issues/97, it's bad practice to place arguments after functions because functions tend to be inlined and eclipse the parameters that follow them.

This could actually be two rules:

- don't allow calling `foo(() => {}, {something: true})`
- don't allow declaring `function foo(cb, options) {}` 

@sindresorhus doesn't like the latter so I'll focus on the first part 😅 

### Fail

```js
element.addEventListener('click', () => {
	// long
}, {once: true})
```

```js
const memoized = memoize(() => {
	// long
}, {
	cache: new WeakMap(),
})
```

### Pass

```js
const handler = () => {
	// long
};
element.addEventListener('click', handler, {once: true})
```

```js
const raw = () => {
	// long
};
const memoized = memoize(raw, {
	cache: new WeakMap(),
})
```

### {maxStatements: 3}

```js
element.addEventListener('click', () => short(), {once: true})
```

```js
const memoized = memoize((x) => alert(x), {
	cache: new WeakMap(),
})
```

### Additional Info

Note that this conflicts with https://github.com/sindresorhus/eslint-plugin-unicorn/blob/main/docs/rules/no-array-callback-reference.md

**Comment by sindresorhus** [comment]
How about allowing one-line functions?

**Comment by fregante** [comment]
What about 

```js
mem(async url => {
  const r = await fetch(url);
  return r.json();
}, {
  something
};
``` 

For the purpose of this rule, this still seems readable. Thats why it should probably default to "3 statements" and warn that "the inline function is too long and impacts readability"

**Comment by sindresorhus** [comment]
I have decided to pass on this

The problem is real for very long inline callbacks, but I don’t think it makes a good general Unicorn rule. The boundary is too subjective. Short callbacks before options are common and readable, for example addEventListener(..., () => open(), {once: true}), setTimeout(() => {}, 100), and array.reduce((a, b) => a + b, 0). A statement threshold would also be arbitrary, since one expression can be unreadable while a few short statements can be fine.

This feels better handled as code review or a project specific no-restricted-syntax rule
